Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+<?php
+/**
+ * MySqlQuery class file.
+ *
+ * @package Freyja\Database
+ * @copyright 2016 SqueezyWeb
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ */
+
+namespace Freyja\Database;
+
+use Freyja\Exceptions\InvalidArgumentException as InvArgExcp;
+use RuntimeException;
+
+/**
+ * MySqlQuery class.
+ *
+ * @package Freyja\Database
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ * @version 1.0.0
+ */
+class MySqlQuery extends Query implements QueryInterface {
+  /**
+   * Query type.
+   *
+   * Accepted values: 'select', 'update', 'insert', 'delete'.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $type;
+
+  /**
+   * Target table.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $table;
+
+  /**
+   * SELECT fields.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $select = array();
+
+  /**
+   * COUNT.
+   *
+   * True correspond to `COUNT(*)`. Or array containing the fields to be
+   * counted. Default false.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var boolean|array
+   */
+  private $count = false;
+
+  /**
+   * ORDERBY.
+   *
+   * Associative array `key => value`, where `key` is the field in which to sort
+   * the results and `value` is the direction.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $order_by = array();
+
+  /**
+   * GROUPBY.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $group_by;
+
+  /**
+   * HAVING clause.
+   *
+   * Array of strings: field name, operator, value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $having = array();
+
+  /**
+   * UPDATE fields and values.
+   *
+   * Associative array `key => value`, where `key` is the fields to update, and
+   * `value` is the new value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $update = array();
+
+  /**
+   * INSERT fields and values.
+   *
+   * Associative array `key => value`, where `key` is the fields to insert, and
+   * `value` is the new value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $insert = array();
+
+  /**
+   * DELETE modifier.
+   *
+   * DELETE modifier can be one of the following: 'LOW_PRIORITY', 'QUICK' and
+   * 'IGNORE'.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $delete_modifier = '';
```

Comment:
It's better to use constants for modifiers, IMO.
Something with binary integer numbers or strings, but constants. And then you save the resulting value in this property.